src/common/java/net/minecraftforge/gradle/common/legacy/LegacyExtension.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.gradle.common.legacy;
+
+import groovy.lang.GroovyObjectSupport;
+import net.minecraftforge.gradle.common.tasks.ExtractZip;
+import net.minecraftforge.gradle.common.util.MinecraftExtension;
+import net.minecraftforge.srgutils.MinecraftVersion;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+
+import java.io.File;
+
+
+/**
+ * Provides an extension block named "legacy".
+ *
+ * <code>
+ * legacy {
+ *     flag = set;
+ * }</code>
+ *
+ * It allows for configuring RetroGradle patches and fixes, which are otherwise set by version.
+ * Each fix is documented in this class and in the application function.
+ * {@link LegacyExtension#runRetrogradleFixes}
+ *
+ * @author Curle
+ */
+public abstract class LegacyExtension extends GroovyObjectSupport {
+    private static final Logger LOGGER = LogManager.getLogger(LegacyExtension.class);
+    public static final String EXTENSION_NAME = "legacy";
+    private static final MinecraftVersion FG3 = MinecraftVersion.from("1.13");
+
+    /**
+     * The RetroGradle project aims to port older (below 1.13) versions of Minecraft to the current toolchains.
+     * In the process, some quirks with the older versions of ForgeGradle were discovered that need to be replicated here.
+     *
+     * Each quirk is documented and accounted for in this function.
+     *  - Classpath / Resources;
+     *      FG2 Userdev puts all classes and resources into a single jar file for FML to consume.
+     *      FG3+ puts classes and resources into separate folders, which breaks on older versions.
+     *      We replicate the FG2 behavior by replacing these folders by the jar artifact on the runtime classpath.
+     *  - Dependency AccessTransformers / Coremods;
+     *      FG2 GradleStart exposes a <code>net.minecraftforge.gradle.GradleStart.csvDir</code> property
+     *      that points to a directory containing CSV mappings files. This is used by LegacyDev to remap dependencies' AT modifiers.
+     *      We replicate the FG2 behavior by extracting the mappings to a folder in the build directory
+     *      and setting the property to point to it.
+     *
+     * This is called from Utils.
+     *
+     * In other words, it's a containment zone for version-specific hacks.
+     * For issues you think are caused by this function, contact Curle or any other Retrogradle maintainer.
```

Comment:
Update documentation to reflect that htis is not a RG project but now part of the official FG

Reply:
On the other hand, the things implemented here are done purely to aid the RG project, someone maintaining this for the benefit of FG only runs the risk of breaking features and fixes required in RG versions.

I'm named there as a maintainer of both.

We should figure out whether this should *stay* as a RG feature in the FG project.